dask/bag/core.py

```diff
@@ -865,11 +866,11 @@ def from_filenames(filenames, chunkbytes=None):
     return Bag(d, name, len(d))
 
 
-def _chunk_read_file(filename, chunkbytes):
+def _chunk_read_file(filename, chunkbytes, encoding):
     extension = os.path.splitext(filename)[1].strip('.')
     compression = {'gz': 'gzip', 'bz2': 'bz2'}.get(extension, None)
 
-    return [(list, (StringIO, (bytes.decode,
+    return [(list, (StringIO, (bytes.decode(encoding),
```

Comment:
This needs to be a callable.  I suggest adding `encoding` as an argument within the tuple.

e.g. replace

    (bytes.decode, (textblock, ...))

with

    (bytes.decode, (textblock, ...), encoding)

This will be like calling the following at execution time.

    bytes.decode(textblock(...), encoding)